include/quic/address.hpp

```diff
@@ -249,6 +250,37 @@ namespace oxen::quic
     template <>
     inline constexpr bool IsToStringFormattable<Address> = true;
 
+    struct RemoteAddress : public Address
+    {
+      private:
+        ustring remote_pubkey;
+
+      public:
+        RemoteAddress() = delete;
+
+        template <typename... Opt>
+        RemoteAddress(std::string_view remote_pk, Opt&&... opts) :
+                Address{std::forward<Opt>(opts)...}, remote_pubkey{to_usv(remote_pk)}
+        {}
+
+        template <typename... Opt>
+        RemoteAddress(ustring_view remote_pk, Opt&&... opts) : Address{std::forward<Opt>(opts)...}, remote_pubkey{remote_pk}
+        {}
+
+        ustring_view view_remote_key() const { return remote_pubkey; }
+        const ustring& get_remote_key() const { return remote_pubkey; }
+
+        RemoteAddress(const RemoteAddress& obj) : Address{*this}, remote_pubkey{obj.remote_pubkey} { _copy_internals(obj); }
```

Comment:
```suggestion
        RemoteAddress(const RemoteAddress& obj) : Address{obj}, remote_pubkey{obj.remote_pubkey} {}
```

Calling `Parent{*this}` in a derived constructor like this is probably undefined behaviour: at this point, the `Address` part of `RemoteAddress` hasn't been initialized yet, and so this ends up constructing `Address` by copying from itself -- even though it hasn't been initialized yet.

But we can just initialize it via copy from `obj` instead, which copies the Address portion out of `obj` into the base `Address` of `this`, and then can just initialize the RemoteAddress portions (thus without needing _copy_internals).

Reply:
yes thats better